RecoEgamma/EgammaTools/plugins/LowPtElectronsModifier.cc

```diff
@@ -0,0 +1,82 @@
+#include "CommonTools/CandAlgos/interface/ModifyObjectValueBase.h"
+#include "DataFormats/BeamSpot/interface/BeamSpot.h"
+#include "DataFormats/VertexReco/interface/VertexFwd.h"
+#include "FWCore/Framework/interface/ConsumesCollector.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "RecoEgamma/EgammaTools/interface/LowPtConversion.h"
+
+////////////////////////////////////////////////////////////////////////////////
+//
+class LowPtElectronModifier : public ModifyObjectValueBase {
+public:
+  LowPtElectronModifier(const edm::ParameterSet& conf, edm::ConsumesCollector&);
+  ~LowPtElectronModifier() override = default;
+
+  void setEvent(const edm::Event&) final;
+  void setEventContent(const edm::EventSetup&) final;
+
+  void modifyObject(pat::Electron& ele) const final;
+
+private:
+  const edm::EDGetTokenT<reco::ConversionCollection> convT_;
+  reco::ConversionCollection* conv_ = nullptr;
+  const edm::EDGetTokenT<reco::BeamSpot> beamSpotT_;
+  reco::BeamSpot* beamSpot_ = nullptr;
+  const edm::EDGetTokenT<reco::VertexCollection> verticesT_;
+  reco::VertexCollection* vertices_ = nullptr;
+  bool extra_;
+};
+
+////////////////////////////////////////////////////////////////////////////////
+//
+LowPtElectronModifier::LowPtElectronModifier(const edm::ParameterSet& conf, edm::ConsumesCollector& cc)
+    : ModifyObjectValueBase(conf),
+      convT_(cc.consumes<reco::ConversionCollection>(conf.getParameter<edm::InputTag>("conversions"))),
+      conv_(),
+      beamSpotT_(cc.consumes<reco::BeamSpot>(conf.getParameter<edm::InputTag>("beamSpot"))),
+      beamSpot_(),
+      verticesT_(cc.consumes<reco::VertexCollection>(conf.getParameter<edm::InputTag>("vertices"))),
+      vertices_(),
+      extra_(conf.getParameter<bool>("addExtraUserVars")) {
+  ;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+//
+void LowPtElectronModifier::setEvent(const edm::Event& iEvent) {
+  conv_ = const_cast<reco::ConversionCollection*>(&iEvent.get(convT_));
```

Comment:
```suggestion
  conv_ = &iEvent.get(convT_);
```
a `const_cast<` in most cases is an indication of something done very wrong.

Reply:
Done.